SofaKernel/modules/Sofa.Geometry/src/sofa/geometry/Tetrahedron.h

```diff
@@ -31,6 +36,31 @@ struct Tetrahedron
     static const sofa::Size NumberOfNodes = 4;
 
     Tetrahedron() = default;
+
+    //needs dot, cross
+    template<typename Node,
+        typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
+        typename = std::enable_if_t<std::is_scalar_v<T>>>
+        static constexpr auto volume(const Node& n0, const Node& n1, const Node& n2, const Node& n3)
+    {
+        constexpr Node n{};
+        //static_assert(std::distance(std::begin(n), std::end(n)) == 3, "volume can only be computed in 3 dimensions.");
+
+        if constexpr (std::distance(std::begin(n), std::end(n)) == 3)
+        {
+            const auto a = n1 - n0;
+            const auto b = n2 - n0;
+            const auto c = n3 - n0;
+
+            return std::abs(sofa::type::dot(sofa::type::cross(a, b), c) / static_cast<T>(6));
+        }
+        else
+        {
+            //does not make sense to compute volume other than 3D
+            //but some code effectively wants 2d volumes(??)
```

Comment:
Maybe a msg_warning? to notify the problem.
And also we could see on the CI where this case happened

Reply:
> 
> 
> which one?

Namely DiagonalMass and MeshMatrixMass (templated on 1d&2d)
Also all the Tetra/HexaSetAlgorithms  templated on 2d...
Maybe some other here and there....

Reply:
> 
> 
> Maybe a msg_warning? to notify the problem. And also we could see on the CI where this case happened

I dont really like adding a message in a "core" function like that.
It should rather be an exception... but this is not deployed yet.
Or the static_assert()  set at the beginning at the function but I commented it because this breaks the compilation for now.
I guess no choice but a temporary message until we can set the static_assert().